src/models/hooks/sessionReportPilot.js

```diff
@@ -130,15 +131,179 @@ const participantsAndNextStepsComplete = async (sequelize, instance, options) =>
+
+      event = data.event;
+      recipients = data.recipients;
+    }
+
+    if (!event?.data?.goal || !event.id) return;
+
+    const eventId = Number(event.id);
+    const eventRecord = await sequelize.models.EventReportPilot.findByPk(eventId, { transaction: options.transaction });
+    if (!eventRecord) throw new Error('Event not found');
+
+    // eslint-disable-next-line no-restricted-syntax
+    for await (const { value: grantValue } of recipients) {
+      const grantId = Number(grantValue);
+
+      // Check if a Goal already exists for the grantId.
+      const existingGoal = await sequelize.models.Goal.findOne({
+        where: { grantId, name: event.data.goal },
+      });
+
+      // If a Goal already exists, do not create a new one and continue with the next recipient.
+      // eslint-disable-next-line no-continue
+      if (existingGoal) continue;
+
+      const grant = await sequelize.models.Grant.findByPk(grantId, { transaction: options.transaction });
+      if (!grant) throw new Error('Grant not found');
+
+      const sessionId = sessionReport.id;
+
+      const hasCompleteSession = await sequelize.models.SessionReportPilot.findOne({
+        where: { eventId, 'data.status': TRAINING_REPORT_STATUSES.COMPLETE },
+        transaction: options.transaction,
+      });
+
+      const status = hasCompleteSession ? 'In Progress' : 'Draft';
+
+      // Since there is no existing Goal, create it and the EventReportPilotGoal record.
+      const newGoal = await sequelize.models.Goal.create({
+        name: event.data.goal,
+        grantId,
+        createdAt: new Date(),
+        updatedAt: new Date(),
+        status,
+        createdVia: 'tr',
+      }, { transaction: options.transaction });
+
+      await sequelize.models.EventReportPilotGoal.create({
+        goalId: newGoal.id,
+        eventId,
+        sessionId,
+        grantId,
+        createdAt: new Date(),
+        updatedAt: new Date(),
+      }, { transaction: options.transaction });
+    }
+  };
+
+  try {
+    if (sequelize.Sequelize && sessionReportOrInstance instanceof sequelize.Sequelize.Model) {
+      await processSessionReport(sessionReportOrInstance);
+    } else {
+      const instance = await sequelize.models.SessionReportPilot.findByPk(sessionReportOrInstance.id, { transaction: options.transaction });
+      if (!instance) throw new Error('SessionReportPilot instance not found');
+      await processSessionReport(instance);
+    }
+  } catch (error) {
+    auditLogger.error(JSON.stringify({ error }));
+  }
+};
+
+const destroyAssociatedGoals = async (sequelize, instance, options) => {
+  try {
+    const { EventReportPilotGoal } = sequelize.models;
+    const goals = await EventReportPilotGoal.findAll({
+      where: {
+        sessionId: instance.id,
+        eventId: instance.eventId,
+      },
+      transaction: options.transaction,
+    });
+
+    // eslint-disable-next-line no-restricted-syntax
+    for await (const goal of goals) {
```

Comment:
I don't think we should destroy Goals here? Won't this destroy goals for other sessions, or goals that could have been used on an AR? So maybe just EventReportPilotGoals. Unless I'm missing something.